core/src/main/java/org/infinispan/remoting/inboundhandler/BaseBlockingRunnable.java

```diff
@@ -28,10 +28,18 @@ protected BaseBlockingRunnable(BasePerCacheInboundInvocationHandler handler, Cac
 
    @Override
    public void run() {
+      InvocationStatus invocationStatus = null;
       try {
-         response = beforeInvoke();
-         if (response == null) {
-            response = handler.invokePerform(command);
+         invocationStatus = beforeInvoke();
+         switch (invocationStatus) {
+            case CONTINUE:
+               response = handler.invokePerform(command);
+               break;
+            case CACHE_NOT_FOUND:
+               response = CacheNotFoundResponse.INSTANCE;
+               break;
+            case DEFERRED:
+               return;
```

Comment:
Not sure I follow (not familiar with this code). (Apart from the fact that I'd like Infinispan to switch to using messages and a channel instead of MessageDispatcher), processing a bunch of messages can be done asynchronously, via the async invocation interface [1]. It is the application's responsibility to enforce ordering at that point, as JGroups considers a message delivered as soon as `receive()` (or `handle()`) returned.

[1] http://www.jgroups.org/manual4/index.html#AsyncInvocation